### DISCUSSION TITLE
Reuse redis client connection in extension

### DISCUSSION THREAD

**Discussion by khaidirsh** [directus/directus#11808] [Archive]
Let's say we are using single redis instance as cache, rate limiting, and application state (another microservice) store. We are going to make endpoint extension to fetch the stored application state and return it to client.

Is it possible to reuse redis client connection that has been used for cache or rate limiting in endpoint extension?

**Comment by rijkvanzanten** [comment]
Not at this point, no. Good feature request though!

**Comment by naskio** [comment]
Any updates on this ?
I would like to use the caching service that is being used by Directus in extensions as well.
That would be great.

**Reply by rijkvanzanten** [reply]
Directus currently relies on multiple redis connections for the various storages, like rate limiting, caching, messaging etc. We're consolidating those into a single redis configuration, at which point we can also pass that redis connection onto extensions 🙂 

https://github.com/directus/directus/issues/15176

**Comment by NilsBaumgartner1994** [comment]
@rijkvanzanten is there a doc how to use the redis store in an extension?

**Comment by Mahoo12138** [comment]
In my opinion, there should be a concept called Service in the extension. For example, an extension provides the Cache Service, and other extensions can use this Service as the Cache. The underlying implementation of the Cache is arbitrary, but it needs to meet the standard interface, which involves the dependency problem of the extension.

**Reply by rijkvanzanten** [reply]
Yes no yes no. When it comes to third party extensions, we cannot guarantee that they won't modify or mess up the system caches, or do something nefarious with the potentially private data kept within. If we make the existing Redis connections available to extensions directly, we'll have to make sure it's somehow sandboxed to ensure the extension can't access any of the system data kept within

**Reply by Mahoo12138** [reply]
That's another question, the Services should be designed with extension isolation.

**Comment by NilsBaumgartner1994** [comment]
Env variable: ENABLE_REDIS_CONNECTION_REUSE_I_KNOW_THE_RISKS = true
